AEROGEAR-8509 - move data sync dashboard form old apb and make it a t…

aliok opened the pull request:
…emplate

Copied from https://github.com/aerogearcatalog/data-sync-apb/blob/master/roles/provision-data-sync-apb/files/data-sync-dashboard.json and replaced the hardcoded app names with `{{ SYNC_APP_NAME }}`

Provisioning using this template is part of another task.


Verification:
* Provovision a sample app with sync-app-apb. Make sure that app has a metrics endpoint.
* Add `org.aerogear.metrics/plain_endpoint: "/metrics"` to the **pod** of the app
* Try this PR file in Grafana. Replace the {{ SYNC_APP_APB }} with the sync app you chose during sync-app-apb provisioning.

I set it up here already: 
* https://grafana-ggg.apps.aliok-64ea.openshiftworkshop.com/dashboard/db/data-sync-test?orgId=1&from=now-5m&to=now
* https://sync-app-test-ggg.apps.aliok-64ea.openshiftworkshop.com/graphql

psturc commented:
@aliok I think the test is failing because of latest changes to `https://hub.docker.com/r/aerogear/voyager-server-example-task` docker image the test is using.
It now requires connection to database and healthcheck endpoint is different.
I think replacing [the image](https://github.com/aerogearcatalog/sync-app-apb/blob/master/vars/test.yml#L3) with [this one](https://hub.docker.com/r/aerogear/voyager-server-example-basic) would fix it.

psturc commented:
👀

aliok commented:
> @aliok I think the test is failing because of latest changes to `https://hub.docker.com/r/aerogear/voyager-server-example-task` docker image the test is using.
It now requires connection to database and healthcheck endpoint is different.
I think replacing [the image](https://github.com/aerogearcatalog/sync-app-apb/blob/master/vars/test.yml#L3) with [this one](https://hub.docker.com/r/aerogear/voyager-server-example-basic) would fix it.

@psturc Changed it, thanks. Also changed the description in the APB param

psturc commented on apb.yml:
```diff
@@ -27,7 +27,7 @@ plans:
         title: Sync Application Name
       - name: SYNC_APP_DOCKER_IMAGE
-        description: URL of your Sync Application Docker Image, e.g. docker.io/aerogear/voyager-server-example-task
+        description: URL of your Sync Application Docker Image, e.g. docker.io/aerogear/voyager-server-example-basic
```
Not sure about this...
@aliok  @wei-lee , do we want to offer here the server app that doesn't use DB configuration (`docker.io/aerogear/voyager-server-example-basic`) or the one that actually requires running db server (`docker.io/aerogear/voyager-server-example-task`).

Or maybe we could update the description and give instructions for both apps?

aliok replied:
I changed it without thinking in detail. I am ok to revert it.
@wei-lee @psturc

psturc replied:
+1 for keeping `voyager-server-example-task`

wei-lee replied:
+1 on keeping the old value

aliok replied:
reverted. merging now.

aliok commented:
@psturc 
> Could any other operation type than `mutation` cause conflicts?
LOL, this is valid question. Gonna get rid of that graph :+1:

aliok commented:
@psturc 

Latest one: https://grafana-ggg.apps.aliok-64ea.openshiftworkshop.com/dashboard/db/data-sync-sync-app-test?panelId=30&orgId=1&from=1549636179678&to=1549636190363&tab=metrics

I did these:
* Removed the conflicts per operation type line graph
* Added some markdown info panel to describe what this dashboard is (it is copied from the mobile-services dashboard. it will be deleted there. see AEROGEAR-8511) 
* Added pod count and version info (similar to above, copied from mobile services dashboard)
